### DISCUSSION TITLE
Integration between FastHistory and tldr-pages

### DISCUSSION THREAD

**Discussion by mkcn** [tldr-pages/tldr#7639] [Show and tell]
Hi everyone,

I just released the 2.4 version of my open-source tool called **fastHistory**, with a new feature which integrates the tldr-pages and I really hope you will appreciate it :blush: 

Project link: https://github.com/mkcn/fastHistory

The tool  does not aim to be a standard TDLR client , but it mainly allows to store your linux command directly in the terminal and quickly find them.  With the new "Discovery" tab I added the possibility to search (by command name or description) also in all TLDR pages, an example of usage is when you need to get the disk space left but you do not remember "df".

![Screenshot from 2022-01-10 23-57-28](https://user-images.githubusercontent.com/7307955/148852014-81f2825d-5ff4-44de-9a0a-4aeb49b52988.png)

Furthermore, to promote collaboration towards your project , with the "CTRL+e" shortcut any user can directly get the TDLR page link of a specific command and  propose an edit in your github repo. 

![Screenshot from 2022-01-10 23-56-24](https://user-images.githubusercontent.com/7307955/148852021-d7a95a3d-80e4-4998-96f6-3a335718eac4.png)

Further details:
- the TLDR project is mentioned also in the [copyright-and-dependencies](https://github.com/mkcn/fastHistory#copyright-and-dependencies) section
- to allow offline mode, the TLDR english pages are downloaded and kept here [https://github.com/mkcn/fastHistory/tree/master/fastHistory/tldr/tldr](https://github.com/mkcn/fastHistory/tree/master/fastHistory/tldr/tldr) 
- only the "linux" and "common" page sections are currently used but it is planned to extend the search to OSX and Windows pages as well (somehow..  someday..)

Any feedback, proposal or bug report is welcome!
